include/capstone/capstone.h

```diff
@@ -290,6 +290,40 @@ typedef struct cs_opt_skipdata {
 	void *user_data;
 } cs_opt_skipdata;
 
+#define MAX_NUM_OP_ENC_ITEMS 8
+#define MAX_NUM_OPC_BITS 64
+
+/// Provides information about an operand's encoding in the instruction
+typedef struct cs_operand_encoding {
+	/// Specifies how many pieces that form the full operand are encoded in the
+	/// instruction separately. For example if count is 2 it means a few bits of
+	/// this operand are in one location and the rest on another. If it's 0 then
+	/// the operand is NOT encoded anywhere in the instruction.
+	uint8_t operand_pieces_count;
+	/// The bit positions of each piece that form the full operand in order. If
+	/// there is only one piece then there is only one index as well. Likewise
+	/// if there are 4 pieces, there are 4 indexes and so on.
+	/// Also note that these indices DON'T necessarily match the indices in the ISA
```

Comment:
I think we should always save bits at the same index as the ISA does. Otherwise there is no reference point for people.